### ISSUE TITLE
Q : does pycrunch has dependency with pycharm auto test?

### ISSUE THREAD

**Issue by raveneer** [gleb-sevruk/pycrunch-engine#21]
1. when i off pycharm's auto test button, pycrunch not run auto.

2. can I modify pycruch code change detecting term sec? (seems little slow)

![image](https://user-images.githubusercontent.com/19799411/91516829-6e322d80-e927-11ea-9c65-d435155be648.png)

**Comment by gleb-sevruk** [comment]
Hi, sorry for late reply.

1. There is no dependency on AutoTest feature from PyCharm. PyCrunch uses more complicated change tracking mechanics, to run tests in affected files, so generally I wouldn't turn AutoTest on while pycrunch is running. But this seems to be an interesting bug, I never tested compatibility.

2. Tests will run as soon as you save the document (Ctrl+S/Command+S), however test discovery will be performed prior to test run. 

Is it slow even when you save document manually? Maybe you have virtual environment *inside* project directory? If you do, I recommend adding it to ignore list using configuration file `.pycrunch-config.yaml`:
```
discovery:
  exclusions:
  # do not search tests inside venv 
  - venv
engine:
  runtime: pytest
```

If this is still slow, maybe there is an issue with FSWatcher or `fs.inotify.max_user_watches` limit. To further diagnose I will need to know your platform (Win/Linux/MacOS), disk type (HHD/SSD) and logs from `pycrunch-engine` output.
Usually on my system it takes < 0.5s to detect changes and start test execution.

And of course you can change code, I will be glad to accept PR that makes change detection faster.

**Comment by gleb-sevruk** [comment]
Maybe you can also check your synchronization options (it may be possible that IDE save file after each typed symbol, and this will result in performance degradation):
![image](https://user-images.githubusercontent.com/609801/91835790-891bdf00-ec52-11ea-9aee-e3394e506d16.png)

**Comment by raveneer** [comment]
sorry for late check. i road mail today.  i will check this ASAP. thank you

**Comment by raveneer** [comment]
added ignore venv and set pycharm's auto test term to 1sec. now test term is short. thank you.

**Comment by gleb-britecore** [comment]
TLDR;

There are no dependencies on PyCharm Test Executor, Pycrunch runs inside it own process and handles test a bit differently.
